### DISCUSSION TITLE
Benchmark IOs Java - Erreur de compilation

### DISCUSSION THREAD

**Discussion by Yoy017** [heig-vd-dai-course/heig-vd-dai-course#371] [Q&A] [answered]
Bonjour, j'essaye de complété le cours `05-java-ios` mais lors de la partie pratique j'ai quelques soucis. J'ai écrit le code pour `BinaryFileWriter` et ai comparé avec la solution fournis. Malgré que la solution est pareil que mon code *(correct donc)*, je reçois l'erreur suivante à la compilation de mon programme:
<img width="602" alt="image" src="https://github.com/user-attachments/assets/1d20511d-5d6a-47f3-9ebe-828a4657650d">

Apparemment il ne trouve pas le fichier Writable malgré son existence dans le répertoire même.

Des pistes sur la cause du problème ?

PS: J'ai tenté de copy past la solution pour voir si je n'ai pas manqué quelque choses et l'erreur survient malgré tout.

**Comment by ludelafo** [comment, marked as answer]
Bonjour Yoann,

Merci pour votre message.

C'est sans doute normal, vous devez compiler votre projet avec Maven.

Pour cela, plusieurs solutions :

- Passer par le terminal et utiliser le wrapper Maven avec la commande `./mvnw clean package`
- Ouvrir le projet dans IntelliJ IDEA et exécuter la "build configuration" nommée "Package the application as JAR file"

Ensuite, une fois votre projet compilé, vous pouvez exécuter l'output à l'aide de la commande `java -jar [...]`.

Les commandes `javac` et `java` permettent de compiler et tester de petits programmes tels que les exemples de code fournis dans le cours car ils se "suffisent à eux-mêmes" (il n'y a pas de dépendances externes). Le contenu pratique est, quant à lui, un projet Maven qui contient des dépendances externes et plusieurs "liens" entre les classes. Il faut donc le compiler avec Maven pour qu'il puisse réunir ces multiples classes en une archive JAR que vous pourrez ensuite exécuter.

Est-ce plus clair maintenant ?

Redonnez-moi des nouvelles !

**Comment by Yoy017** [comment]
Oui, merci pour votre réponse. Maintenant j'arrive à exécuter les différentes lignes de commandes du projet !😄

**Reply by ludelafo** [reply]
Super, merci pour le retour !
